sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala

```diff
@@ -823,6 +823,8 @@ private[columnar] object ColumnType {
       case DoubleType => DOUBLE
       case StringType => STRING
       case BinaryType => BINARY
+      case YearMonthIntervalType => YEAR_MONTH_INTERVAL
+      case DayTimeIntervalType => DAY_TIME_INTERVAL
```

Comment:
Merge this to L820-821?

Reply:
https://github.com/apache/spark/pull/32452#discussion_r629887096